ubccr/xdmod
# Set version number dynamically for REST API documentation page.

## Description



This makes it so that when the static site is generated, the version number on the REST API documentation page is set to match the software version number.

## Motivation and Context


We forgot to update the version number on the REST API documentation page from 10.0.0. This uses an existing variable in the Jekyll static site generator configuration so that we do not have to redundantly specify it or remember to manually update it. Since the REST API is versioned concurrently with Open XDMoD, the current REST API version should be the same as the current Open XDMoD version.

## Tests performed



I followed these steps on my laptop:
1. [Make sure Jekyll is installed](https://jekyllrb.com/docs/installation/).
1. Change to the `docs` directory.
1. Run these commands:
    ```
    git clone https://github.com/ubccr/xdmod-jekyll-theme
    bundle install
    bundle add webrick
    bundle exec jekyll serve
    ```
1. In Chrome, browse to the server address listed in the terminal and confirm the page looks correct.
1. In Chrome, Save As -> Webpage, Single File -> save as a temporary file `new.mht`.
1. Revert the changes to `docs/xdmod-rest-schema.json` and let the site regenerate.
1. In Chrome, refresh, confirm the version number is back to 10.0.0, and Save As -> Webpage, Single File -> save as a temporary file `old.mht`.
1. Diff `old.mht` and `new.mht` and confirm the only difference besides random IDs is the version number has been updated to `11.0.0`.

## Checklist:
